app/main.py

```diff
@@ -1 +1,37 @@
-# write your code here
+class Animal:
+    alive = []
+
+    def __init__(
+            self,
+            name: str,
+            health: int = 100,
+            hidden: bool = False
+    ) -> None:
+        self.name = name
+        self.health = health
+        self.hidden = hidden
+        self.alive.append(self)
+
+    def __repr__(self) -> str:
+        return (f"{{Name: {self.name}, "
+                f"Health: {self.health}, "
+                f"Hidden: {self.hidden}}}"
+                )
+
+
+class Herbivore(Animal):
+
+    def hide(self) -> None:
+        self.hidden = not self.hidden
+
+
+class Carnivore(Animal):
+
+    def bite(self, herbivore_animal: Herbivore) -> None:
+        if (not herbivore_animal.hidden
```

Comment:
Simplify this code:
Shorten parameter `herbivore_animal `to just `herbivore` and use class name `Animal `instead of `self`. Then
```
if isinstance(herbivore, Herbivore) and not herbivore.hidden:
    herbivore.health -= 50
    if herbivore.health <= 0:
        Animal.alive.remove(herbivore)
```